compiler/rustc_hir_analysis/src/hir_ty_lowering/cmse.rs

```diff
@@ -67,13 +70,13 @@ pub(crate) fn validate_cmse_abi<'tcx>(
 /// Returns whether the inputs will fit into the available registers
 fn is_valid_cmse_inputs<'tcx>(
     tcx: TyCtxt<'tcx>,
-    fn_sig: ty::PolyFnSig<'tcx>,
+    fn_sig: ty::FnSig<'tcx>,
 ) -> Result<Result<(), usize>, &'tcx LayoutError<'tcx>> {
     let mut span = None;
     let mut accum = 0u64;
 
-    for (index, arg_def) in fn_sig.inputs().iter().enumerate() {
-        let layout = tcx.layout_of(ParamEnv::reveal_all().and(*arg_def.skip_binder()))?;
+    for (index, ty) in fn_sig.inputs().iter().enumerate() {
+        let layout = tcx.layout_of(ParamEnv::reveal_all().and(*ty))?;
```

Comment:
Side-note: this param-env is not right, so it'll probably ICE in other cases...

Reply:
yeah, https://github.com/rust-lang/rust/issues/130104 was just reported and is not fixed by this branch (currently). How do we get a correct `ParamEnv` here?

Reply:
here https://github.com/rust-lang/rust/pull/127814#discussion_r1680842560 is where some discussion around the current design happened. Is there an existing way to perform that traversal after type checking to perform this layout check?